activemq-client/src/main/java/org/apache/activemq/command/ActiveMQMapMessage.java

```diff
@@ -827,4 +839,26 @@ public Map<String, Object> getContentMap() throws JMSException {
         initializeReading();
         return map;
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean isBodyAssignableTo(Class c) throws JMSException {
+        final Map<String, Object> map = getContentMap();
+        if (map == null || map.isEmpty()) {
```

Comment:
doGetBody doesnt seem to special case the empty map as this does...should it? (since it special casing saying true while being given any class here)

Reply:
Good catch, it likely should because the map message impl treats an empty map as basically null (no body) so if it's empty we should return null I think and not an empty map.